Feature educational programs

olmyk opened the pull request:
Add functionality for Educational Programs & Ag Grid tables columns autoresize

olmyk committed:
add EducationalPrograms

KolyaNosenko commented:
Давай не використовувати скорочення EP, оскільки це не загальнозрозуміла абревіатура. Краще писати повністю “EducationalProgram”

KolyaNosenko commented on src/educational-programs/ui/components/EditEPModal/EditEPModal.tsx:
```diff
@@ -0,0 +1,127 @@
+  return (
+    <Modal open={open} onClose={onClose}>
+      <Content>
+        <Title>Room information</Title>
```
Room?

KolyaNosenko commented on src/educational-programs/ui/components/EPTable/EPTable.tsx:
```diff
@@ -0,0 +1,152 @@
+  const seenStudyFieldIds = new Set<number>();
+  const seenSpecialtyIds = new Set<number>();
+
+  for (const sf of studyFieldMap.values()) {
```
Складно виглядає(

KolyaNosenko commented on src/educational-programs/ui/components/EPTable/EPTable.tsx:
```diff
@@ -0,0 +1,152 @@
+const EPTable = ({ data, onEPsSelected }: EPTableProps) => {
+  const gridRef = useRef<AgGridReact<FlatEducationalProgram>>(null);
+
+  const flatData = groupEducationalPrograms(data);
```
Додай useMemo

KolyaNosenko commented on src/educational-programs/ui/components/EPToolbar/EPToolbar.tsx:
```diff
@@ -0,0 +1,54 @@
+    <Root className={className}>
+      <ActionsList>
+        <ActionsListItem>
+          <AddButton onClick={onAdd}>Add new room</AddButton>
```
Room?

KolyaNosenko commented on src/educational-programs/ui/components/EditEPModal/EditEPModal.tsx:
```diff
@@ -0,0 +1,127 @@
+            <InputLabel shrink htmlFor="educational-program-degree-type">
+              Degree Type
+            </InputLabel>
+            <FormTextField
```
Тут точно все string? Тут, напевно, треба select робити, подивись, як на Lecturers

olmyk committed:
change all occurrences of EP to EducationalProgram

olmyk committed:
fixes regarding PR#19 review

fix Rooms text misuse,
update column types to Select,
update groupEducationalPrograms()

olmyk commented:
Оновив
